src/utils.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+
+// custom react hook to enable and connect users Marina account.
+function useChecks() {
+  const [isInstalled, setIsInstalled] = useState(false);
+  const [isConnected, setIsConnected] = useState(false);
+
+  useEffect(() => {
+    let isCheckingMarina = false;
+    const interval = setInterval(async () => {
+      try {
+        const marina = window.marina;
+
+        if (marina === undefined) return;
+
+        setIsInstalled(true);
+
+        if (isCheckingMarina) return;
+        isCheckingMarina = true;
+
+        let isEnabled = await window.marina.isEnabled();
+
+        if (isEnabled) {
```

Comment:
Wouldn't be better to "siwtch" inside the button the onClick callback, instead of imperatively setup it here?